### ISSUE TITLE
[Potential addition to the CLI] Ability to re-use an existing report, if we have one in storage from another command

### ISSUE THREAD

**Issue by vlad-ko** [codecov/feedback#229]
**What product do you want to improve?**
General processing of the reports by Codecov 

**Is your feature request related to a problem? Please describe.**
One of our customers has the following workflow:

1. PR is opened
2. All affected subprojects are built and unit tests run (this report has all the required coverage info to establish a solid base)
3. Coverage report uploaded
4. PR merged
5. All affected subprojects are built and unit tests run
6. Coverage report uploaded

They are trying to cut down on the redundancy of 5/6 since the coverage data is the exact same before the merge happens, and we haven't had issues with stale coverage data so far

**Describe the solution you'd like**
Add a command to `codecov-cli` that would do something like "re-use" report from <SHA>. This would save the customer the need to run tests and do an upload, and will simply re-use the existing report from storage.
I know we have an `empty-upload` command, but it seems to serve a slightly different purpose.

**Describe alternatives you've considered**
Leverage `empty-upload` command, but make sure we re-run all of the required calculations and can essentially establish a new base from this upload. 

**Additional context**
To summarize what they really want is:

1. Some way to fetch the coverage report in step 3,
2. Some way to do step 6 without having to use the Codecov upload binary or the codecov-cli
